helix-core/src/main/java/org/apache/helix/spectator/RoutingTableProvider.java

```diff
@@ -558,23 +558,30 @@ protected void refresh(Map<String, Map<String, Map<String, CurrentState>>> curre
 
   private void resetRoutingTableAndNotify(long startTime, RoutingTable newRoutingTable) {
     _routingTableRef.set(newRoutingTable);
-    logger.info("Refresh the RoutingTable for cluster {}, takes {} ms.",
-        (_helixManager != null ? _helixManager.getClusterName() : null),
+    String clusterName = _helixManager != null ? _helixManager.getClusterName() : null;
+    logger.info("Refresh the RoutingTable for cluster {}, takes {} ms.", clusterName,
         (System.currentTimeMillis() - startTime));
-    notifyRoutingTableChange();
+
+    // TODO: move the callback user code logic to separate thread upon routing table statePropagation latency
+    // integration test result. If the latency is more than 2 secs, we need to change this part.
```

Comment:
Question: Which test are you specifically referring to? Do we have an integration test that fails when the latency exceeds 2 secs? Is 2 secs an arbitrary number?

Reply:
Yes, storage team is doing integration test and they are blocked by this one. 2 sec is perceived number from meeting with them